### PR TITLE
(Changed) -  Renamed PaymentService and connected PX_ENVIRONMENT

### PR DESCRIPTION
**Pull request by jscaramal-meli**
## What have changed?

- Renamed PaymentService to CheckoutService

- Connected PX_ENVIRONMENT with RequestInfos for environment selection

## How to test:

Set PX_ENVIRONMENT on Info.plist->Information property list, choose between [alpha, beta, gamma, prod] and check for proper API calls

### CODE REVIEW

File: MercadoPagoSDK/MercadoPagoSDK/RequestLayer/Checkout/CheckoutService.swift [hunk 1]
```diff
@@ -1,25 +1,25 @@
-protocol PaymentService {
+protocol CheckoutService {
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/RequestLayer/Checkout/CheckoutService.swift:1]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/RequestLayer/Checkout/CheckoutService.swift#L1"></span>A 'main_type' should not be placed amongst the file type(s) 'supporting_type'.<br />
<code>file_types_order</code> <code>CheckoutService.swift:1</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift [hunk 2]
```diff
@@ -63,14 +63,13 @@ internal class MercadoPagoServices: NSObject {
         }
     }
 
-    func getOpenPrefInitSearch(pref: PXCheckoutPreference, cardsWithEsc: [String], oneTapEnabled: Bool, splitEnabled: Bool, discountParamsConfiguration: PXDiscountParamsConfiguration?, flow: String?, charges: [PXPaymentTypeChargeRule], headers: [String: String]?,  newCardId: String?, callback : @escaping (PXInitDTO) -> Void, failure: @escaping ((_ error: PXError) -> Void)) {
-        
+    func getOpenPrefInitSearch(pref: PXCheckoutPreference, cardsWithEsc: [String], oneTapEnabled: Bool, splitEnabled: Bool, discountParamsConfiguration: PXDiscountParamsConfiguration?, flow: String?, charges: [PXPaymentTypeChargeRule], headers: [String: String]?, newCardId: String?, callback : @escaping (PXInitDTO) -> Void, failure: @escaping ((_ error: PXError) -> Void)) {
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift:66]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift#L66"></span>Function should have 5 parameters or less: it currently has 11<br />
<code>function_parameter_count</code> <code>MercadoPagoServices.swift:66</code></td>
    </tr>
  </tbody>
</table>

Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift:66]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift#L66"></span>Line should be 200 characters or less: currently 376 characters<br />
<code>line_length</code> <code>MercadoPagoServices.swift:66</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift [hunk 3]
```diff
@@ -63,14 +63,13 @@ internal class MercadoPagoServices: NSObject {
...
         let bodyFeatures = PXInitFeatures(oneTap: oneTapEnabled, split: splitEnabled)
         let body = PXInitBody(preference: pref, publicKey: publicKey, flow: flow, cardsWithESC: cardsWithEsc, charges: charges, discountConfiguration: discountParamsConfiguration, features: bodyFeatures, newCardId: newCardId, checkoutType: checkoutType)
 
         let bodyJSON = try? body.toJSON()
 
-        paymentService.getInit(preferenceId: nil, privateKey: privateKey, body: bodyJSON, headers: headers) { apiResponse in
+        checkoutService.getInit(preferenceId: nil, privateKey: privateKey, body: bodyJSON, headers: headers) { apiResponse in
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift:72]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift#L72"></span>Line should be 120 characters or less: currently 125 characters<br />
<code>line_length</code> <code>MercadoPagoServices.swift:72</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift [hunk 4]
```diff
@@ -84,7 +83,7 @@ internal class MercadoPagoServices: NSObject {
 
         let bodyJSON = try? body.toJSON()
 
-        paymentService.getInit(preferenceId: preferenceId, privateKey: privateKey, body: bodyJSON, headers: headers) { apiResponse in
+        checkoutService.getInit(preferenceId: preferenceId, privateKey: privateKey, body: bodyJSON, headers: headers) { apiResponse in
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift:86]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift#L86"></span>Line should be 120 characters or less: currently 134 characters<br />
<code>line_length</code> <code>MercadoPagoServices.swift:86</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift [hunk 5]
```diff
@@ -101,7 +100,7 @@ internal class MercadoPagoServices: NSObject {
         }
     }
 
-    func getPointsAndDiscounts(url: String, uri: String, paymentIds: [String]? = nil, paymentMethodsIds: [String]? = nil, campaignId: String?, prefId: String?, platform: String, ifpe: Bool, merchantOrderId: Int?, headers: [String: String], paymentTypeId: String?,callback : @escaping (PXPointsAndDiscounts) -> Void, failure: @escaping (() -> Void)) {
+    func getPointsAndDiscounts(url: String, uri: String, paymentIds: [String]? = nil, paymentMethodsIds: [String]? = nil, campaignId: String?, prefId: String?, platform: String, ifpe: Bool, merchantOrderId: Int?, headers: [String: String], paymentTypeId: String?, callback : @escaping (PXPointsAndDiscounts) -> Void, failure: @escaping (() -> Void)) {
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift:103]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift#L103"></span>Line should be 200 characters or less: currently 351 characters<br />
<code>line_length</code> <code>MercadoPagoServices.swift:103</code></td>
    </tr>
  </tbody>
</table>

Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift:103]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDK/Services/MercadoPagoServices.swift#L103"></span>Function should have 5 parameters or less: it currently has 11<br />
<code>function_parameter_count</code> <code>MercadoPagoServices.swift:103</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift [hunk 6]
```diff
@@ -25,84 +25,83 @@ class CustomServiceMock: CustomService {
...
-    
+
     func resetESCCap(cardId: String, privateKey: String?, response: @escaping (Swift.Result<Void, PXError>) -> Void) {
         calledResetESCCap = true
         successResponse ? response(.success(())) : response(.failure(PXError(domain: "", code: 0)))
     }
-    
-    func createPayment(privateKey: String?, publicKey: String, data: Data?, header: [String : String]?, response: @escaping (Swift.Result<PXPayment, PXError>) -> Void) {
+
+    func createPayment(privateKey: String?, publicKey: String, data: Data?, header: [String: String]?, response: @escaping (Swift.Result<PXPayment, PXError>) -> Void) {
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift:34]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift#L34"></span>Line should be 120 characters or less: currently 168 characters<br />
<code>line_length</code> <code>MercadoPagoServicesTest.swift:34</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift [hunk 7]
```diff
@@ -25,84 +25,83 @@ class CustomServiceMock: CustomService {
...
 }
 
-class PaymentServicesMock: PaymentService {
+final class CheckoutServicesMock: CheckoutService {
     var successResponse = true
     var calledGetInit = false
-    
-    func getInit(preferenceId: String?, privateKey: String?, body: Data?, headers: [String : String]?, completion: @escaping (Swift.Result<PXInitDTO, PXError>) -> Void) {
+
+    func getInit(preferenceId: String?, privateKey: String?, body: Data?, headers: [String: String]?, completion: @escaping (Swift.Result<PXInitDTO, PXError>) -> Void) {
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift:76]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift#L76"></span>Line should be 120 characters or less: currently 169 characters<br />
<code>line_length</code> <code>MercadoPagoServicesTest.swift:76</code></td>
    </tr>
  </tbody>
</table>

File: MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift [hunk 8]
```diff
@@ -25,84 +25,83 @@ class CustomServiceMock: CustomService {
...
-class MercadoPagoServicesTest: XCTestCase {
+final class MercadoPagoServicesTest: XCTestCase {
     var sut: MercadoPagoServices!
 
     var customService: CustomServiceMock!
     var remedyService: RemedyServicesMock!
     var gatewayService: GatewayServicesMock!
-    var paymentService: PaymentServicesMock!
-
+    var checkoutService: CheckoutServicesMock!
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift:88]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/23442c6d9133eb427e3b23234563ca2e3cd5c832/MercadoPagoSDK/MercadoPagoSDKTests/MercadoPagoServicesTest.swift#L88"></span>Implicitly unwrapped optionals should be avoided when possible.<br />
<code>implicitly_unwrapped_optional</code> <code>MercadoPagoServicesTest.swift:88</code></td>
    </tr>
  </tbody>
</table>

**Comment by pedro-nadolny** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The PR title should follow the title convetion: (Added,Changed,Deprecated,Removed,Fixed,Security)[JIRA-XXX] - (brief description here)</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">SwiftLint also found 16 more violations with this PR.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR does not include any JIRA tasks in the title. (e.g. [TICKET-1234])</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">You and the size of your PR are awesome! 🚀</td>
    </tr>
  </tbody>
</table>

## Current coverage for MercadoPagoSDKV4.framework is `2.89%`
Files changed | - | - 
--- | --- | ---
RequestInfos.swift | `0.00%` | :skull:
MercadoPagoServices.swift | `63.05%` | :warning:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_danger_pr">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by mvarani — APPROVED** [review]

File: MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/RequestInfos.swift [hunk 9]
```diff
@@ -7,13 +7,14 @@ enum HTTPMethodType: String {
...
 
-enum BackendEnvironment: String {
+enum BackendEnvironment: String, CaseIterable {
     case alpha = "alpha/"
     case beta = "beta/"
     case prod = "production/"
     case gamma = "gamma/"
 }
 
+
```
Comment (pedro-nadolny) [inline MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/RequestInfos.swift:17]: <table data-meta="generated_by_danger_pr">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/mercadopago/px-ios/blob/bda2feb146f7ac50f96f099e343e498ff46019d8/MercadoPagoSDK/MercadoPagoSDK/RequestLayer/RequestStruct/RequestInfos.swift#L17"></span>Limit vertical whitespace to a single empty line. Currently 2.<br />
<code>vertical_whitespace</code> <code>RequestInfos.swift:17</code></td>
    </tr>
  </tbody>
</table>

**Pull request merged**